lib/crystalball/rspec/runner.rb

```diff
@@ -109,6 +109,46 @@ def prediction_example_groups(out)
 
         pruner.pruned_groups
       end
+
+      def setup(err, out)
+        configure(err, out)
+        @configuration.load_spec_files
+
+        # Since prediction compacting is disabled we need to remove filtering
+        # in cases like './spec/foo.rb ./spec/foo.rb[1:1:2]'
+        remove_unecessary_filters(@options.options[:files_or_directories_to_run])
+
+        @world.announce_filters
+      end
+
+      def configure(err, out)
+        @configuration.error_stream = err
+        @configuration.output_stream = out if @configuration.output_stream == $stdout
+        @options.configure(@configuration)
+      end
+
+      private
+
+      def remove_unecessary_filters(files_or_directories)
+        directories, files = files_or_directories.partition { |f| File.directory?(f) }
+        remove_unecessary_filters_from_files(files)
+        remove_unecessary_filters_from_directories(directories)
+      end
+
+      def remove_unecessary_filters_from_directories(directories)
+        directories.each do |dir|
+          files = @configuration.__send__(:gather_directories, dir)
+          remove_unecessary_filters_from_files(files)
+        end
+      end
+
+      def remove_unecessary_filters_from_files(files)
+        files.select { |f| ::RSpec::Core::Example.parse_id(f).last.nil? }.each do |file|
+          next remove_unecessary_filters(fd) if File.directory?(file)
+          path = ::RSpec::Core::Metadata.relative_path(File.expand_path(file))
+          @configuration.filter_manager.inclusions[:ids].delete(path)
+        end
+      end
```

Comment:
Lets move it out to separate class. Expected usage in `setup`:
```ruby
Crystalball::RSpec::Filtering.remove_unnecessary!(@configuration, @options.options[:files_or_directories_to_run])
```
and please add extensive description of why we need this functionality.